addons/mail/static/tests/helpers/mock_server.js

```diff
@@ -2072,7 +2066,28 @@ MockServer.include({
         return false;
     },
     /**
-     * Simulates `get_display_value` on `mail.tracking.value`
+     * Simulates `_tracking_value_format` on `mail.tracking.value`
+     */
+    _mockMailTrackingValue_TrackingValueFormat(tracking_value_ids) {
+        const tracking_values = tracking_value_ids.map(tracking => ({
+            changedField: tracking.field_desc,
+            id: tracking.id,
+            newValue: [['insert-and-replace', {
+                fieldType: tracking.field_type,
+                value: this._mockMailTrackingValue_GetDisplayValue(tracking, 'new')
+            }]],
+            oldValue: [['insert-and-replace', {
+                fieldType: tracking.field_type,
+                value: this._mockMailTrackingValue_GetDisplayValue(tracking, 'old')
+            }]],
+        }));
+        if (!tracking_values) {
+            return [];
+        }
```

Comment:
Condition is useless: 1. it should do insert-and-replace empty instead so it can be removed, 2. check for empty array in js needs length

Reply:
`[('insert-and-replace', [])]` will delete all the records and should always be avoided, which is the case. I will do the length check in this case.